### PR TITLE
Add Java benchmarks

### PR DESCRIPTION
**Pull request by marcospereira**
1. A minimal Java app without default filters
2. Form and file upload
3. Update play-json

### CODE REVIEW

**Comment by marcospereira** [comment]
@rich this also adds raw upload endpoints for Java tests.

File: java-di-bench/app/controllers/BenchController.java [hunk 1]
```diff
@@ -50,9 +65,23 @@ public Result downloadChunked(final int length) {
...
+
+    String filename = uploadedFile.getFilename();
+    File source = uploadedFile.getFile();
+    // Do nothing with the file.
+    return ok("It works!");
+  }
+
+  @BodyParser.Of(BodyParser.Raw.class)
+  public Result uploadRaw() {
+    return ok("It works"); // TODO: Verify upload happened
```
Comment (richdougherty) [inline java-di-bench/app/controllers/BenchController.java]: > `// TODO: Verify upload happened`

Just check the raw file length?

File: java-di-bench/app/controllers/BenchController.java [hunk 2]
```diff
@@ -69,4 +98,12 @@ public Result jsonEncode() {
     return ok(result);
   }
 
+  public Result simpleForm() {
+    Form<User> userForm = formFactory.form(User.class).bindFromRequest();
+    if (userForm.hasErrors()) {
+      return badRequest("This shouln't happen");
```
Comment (richdougherty) [inline java-di-bench/app/controllers/BenchController.java]: Typo `shouln't` → `shouldn't`

**Review by richdougherty — APPROVED** [review]

File: java-di-bench/app/controllers/BenchController.java (hunk 1 above)
Comment (marcospereira) [reply java-di-bench/app/controllers/BenchController.java]: Actually, I don't think we need anything here, just like we do in the Scala benchs.

File: java-di-bench/app/controllers/BenchController.java (hunk 2 above)
Comment (marcospereira) [reply java-di-bench/app/controllers/BenchController.java]: Fixed. ;-)

**Pull request merged**